OneDrive-Cloud-Player/Views/MainPage.xaml.cs

```diff
@@ -1,12 +1,34 @@
-﻿using Windows.UI.Xaml.Controls;
+﻿using Microsoft.Extensions.DependencyInjection;
+using OneDrive_Cloud_Player.ViewModels;
+using Windows.UI.Xaml.Controls;
+using Windows.UI.Xaml.Navigation;
 
 namespace OneDrive_Cloud_Player.Views
 {
-    public sealed partial class MainPage : Page
+    public sealed partial class MainPage
     {
         public MainPage()
         {
             this.InitializeComponent();
+            var container = App.Current.Container;
+            DataContext = ActivatorUtilities.GetServiceOrCreateInstance(container, typeof(MainPageViewModel));
+        }
+
+        protected override void OnNavigatingFrom(NavigatingCancelEventArgs e)
```

Comment:
maybe a clarification or comment about when this method is called, since we were confused about it before